dissect/target/plugins/os/unix/log/messages.py

```diff
@@ -48,68 +42,65 @@ def messages(self):
-                        except Exception:
-                            pid = None
-
-                        try:
-                            msg = RE_MSG.search(line).groups()[0]
-                            if msg is None:
-                                msg = line
-                        except Exception:
-                            msg = line
-
-                        yield MessagesRecord(
-                            ts=ts,
-                            filepath=f.path,
-                            daemon=daemon,
-                            pid=pid,
-                            msg=msg,
-                            raw=line,
-                            _target=self.target,
-                        )
+
+        log_files: [TargetPath] = list(self.target.fs.path("/var/log/").glob("syslog*")) + list(
+            self.target.fs.path("/var/log/").glob("messages*")
+        )
+
+        for log_file in log_files:
+
+            file_ctime = self.target.fs.get(str(log_file)).stat().st_ctime
+            year_file_created = datetime.fromtimestamp(file_ctime).year
+            last_seen_year = year_file_created
+            last_seen_month = 0
+
+            for line in decompress_and_readlines(log_file):
+
+                # Line can be a byte object on debian and centos
+                if isinstance(line, bytes):
+                    line = line.decode("utf-8").strip()
+                else:
+                    line = line.strip()
+
+                if not line or line == "":
+                    continue
+
+                try:
+                    ts = datetime.strptime(RE_TS.search(line).groups()[0], "%b %d %H:%M:%S")
+                    if last_seen_month > ts.month:
+                        last_seen_year += 1
+                    last_seen_month = ts.month
+                    ts = ts.replace(year=last_seen_year)
+                except Exception as e:
+                    self.target.log.warn(f"Could not convert timestamp line in {log_file}: {e}")
+                    # Set ts to epoch 1970-01-01 if we could not convert
+                    ts = datetime.utcfromtimestamp(0)
+
+                try:
+                    daemon = RE_DAEMON.search(line).groups()[0]
+                except Exception:
+                    daemon = None
+
+                try:
+                    pid = RE_PID.search(line)
+                    if pid is not None:
+                        pid = pid.groups()[0]
+                except Exception:
+                    pid = None
+
+                try:
+                    msg = RE_MSG.search(line).groups()[0]
+                    if msg is None:
+                        msg = line
+                except Exception:
+                    msg = line
+
+                yield MessagesRecord(
+                    ts=ts,
+                    daemon=daemon,
+                    pid=pid,
+                    message=msg,
+                    raw=line,
+                    source=str(log_file),
```

Comment:
```suggestion
                    source=path.from_posix(log_file),
```
And add `from flow.record.fieldtypes import path`.